app/scripts/controllers/threebox.js

```diff
@@ -32,16 +32,18 @@ class ThreeBoxController {
         if (origin !== '3Box') { return [] }
         const isUnlocked = getKeyringControllerState().isUnlocked
 
-        const selectedAddress = getSelectedAddress()
+        const accounts = await this.keyringController.getAccounts()
 
-        if (isUnlocked && selectedAddress) {
-          return [selectedAddress]
+        if (isUnlocked && accounts[0]) {
+          const appKeyAddress = await this.keyringController.getAppKeyAddress(accounts[0], 'wallet://3box.metamask.io')
+          return [appKeyAddress]
         } else {
           return []
         }
       },
-      processPersonalMessage: (msgParams) => {
-        return Promise.resolve(keyringController.signPersonalMessage(msgParams, {
+      processPersonalMessage: async (msgParams) => {
+        const accounts = await this.keyringController.getAccounts()
+        return Promise.resolve(keyringController.signPersonalMessage({ ...msgParams, from: accounts[0] }, {
```

Comment:
Nit: `Promise.resolve(...)` is redundant here, as `async` functions automatically wrap non-Promise return values